src/withStyles.jsx

```diff
@@ -67,7 +70,23 @@ export function withStyles(
             }}
           />
         );
-      }
+      };
+    } else {
+      // WrappedComponent is a function, so we can avoid a little overhead by
+      // calling it directly instead of rendering it. This avoids React
+      // lifecycle overhead for the wrapped component and may not be necessary
+      // in a future version of React.
+      WithStyles.prototype.render = function render() {
+        const { themeName } = this.context;
+        flushIfNecessary();
+        return WrappedComponent({
+          ...this.props,
+          ...{
+            [themePropName]: ThemedStyleSheet.get(themeName),
+            [stylesPropName]: styleDef ? styleDef(themeName) : EMPTY_STYLES,
+          },
+        }, this.context);
```

Comment:
i think you need to actually spread in `WrappedComponent.defaultProps` in here too, since that's normally what `<WrappedComponent {...this.props} />` would be doing